R/prep_and_calc_loan_book.R

```diff
@@ -103,17 +103,14 @@ run_prep_calculation_loans <- function(credit_type = "outstanding") {
   )
 
   #### Wrangle and prepare data-------------------------------------------------
-
-  # TODO: what to do with negative credit limits?
-  matched_non_negative <- matched %>%
-    dplyr::mutate(
-      loan_size_outstanding = dplyr::if_else(
-        .data$loan_size_outstanding < 0, 0, .data$loan_size_outstanding
-      ),
-      loan_size_credit_limit = dplyr::if_else(
-        .data$loan_size_credit_limit < 0, 0, .data$loan_size_credit_limit
-      )
-    )
+  # ADO 2690 - remove rows with negative loan values (not allowed in P4B)
+  if (credit_type == "outstanding") {
+    matched_non_negative <- matched %>%
+      dplyr::filter(.data$loan_size_outstanding >= 0)
+  } else {
+    matched_non_negative <- matched %>%
+      dplyr::filter(.data$loan_size_credit_limit >= 0)
```

Comment:
just to make sure, since in old version you corrected negative values for both credit_types. Is it the case that the column of the not selected credit_type  is of no relevance?

Reply:
yes, I think this is actually safer, because only one is of relevance for the further calculation. Thus, if for whatever reason, the other column is negative, this should not be removed